Makefile

```diff
@@ -156,21 +156,33 @@ build/%:
 	mkdir -p $@
 
 build/$(ASPROF): src/main/* src/jattach/* src/fdtransfer.h
+ifeq ($(BINARIES_DIRECTORY),)
```

Comment:
These conditions make build flow more branchy and harder to follow, yet they don't save from redundant build steps like creating CONVERTER_JAR.

What about creating a separate target for running Java tests against user-provided binaries?